version: "3.8"
services:
  account_api:
    build: ./account-api
    command: python manage.py run -h 0.0.0.0
    volumes:
      - ./account-api/:/app
    ports:
      - 5001:5000
    environment:
      - FLASK_APP=init/__init__.py
      - FLASK_DEBUG=1
      - DATABASE_URL=postgresql://user:password@account_db_dev:5432/account_db_dev
    depends_on:
      - account_db_dev
    networks:
      - dev
  account_db_dev:
    image: postgres:15-alpine
    volumes:
      - pgaccount-data-dev:/var/lib/postgresql/data/
    environment:
      - POSTGRES_USER=user
      - POSTGRES_PASSWORD=password
      - POSTGRES_DB=account_db_dev
    ports:
      - "5432:5432"
    networks:
      - dev
networks:
  dev:
    driver: bridge
volumes:
  pgaccount-data-dev:
